  INFO root [2020-11-08 21:29:56,192] - 
  INFO root [2020-11-08 21:29:56,196] - -----------------------------------------------------------------------------------------


  INFO root [2020-11-08 21:29:56,199] - Resmgr::initialize: KBE_ROOT=D:/kbengine_24pic/
  INFO root [2020-11-08 21:29:56,199] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine_24pic/kbe/res/;D:/kbengine_24pic/kbengine_stone_assets/;D:/kbengine_24pic/kbengine_stone_assets/scripts/;D:/kbengine_24pic/kbengine_stone_assets/res/
  INFO root [2020-11-08 21:29:56,200] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine_24pic/kbe/bin/server/
  INFO root [2020-11-08 21:29:56,220] - NetworkInterface::initialize(EXTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:20013 )
  INFO root [2020-11-08 21:29:56,236] - NetworkInterface::initialize(EXTERNAL): address 192.168.2.246:20013, SOMAXCONN=128.
  INFO root [2020-11-08 21:29:56,241] - NetworkInterface::initialize(INTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:51765 )
  INFO root [2020-11-08 21:29:56,242] - NetworkInterface::initialize(INTERNAL): address 192.168.2.246:51765, SOMAXCONN=128.
  INFO root [2020-11-08 21:29:56,244] - server-configs:
  INFO root [2020-11-08 21:29:56,244] - 	internalAddr : 192.168.2.246:51765
  INFO root [2020-11-08 21:29:56,244] - 	externalAddr : 192.168.2.246:20013
  INFO root [2020-11-08 21:29:56,244] - 	componentID : 920002
  INFO root [2020-11-08 21:29:56,649] - Components::findLogger: found logger, addr:192.168.2.246:51760
  INFO root [2020-11-08 21:29:56,650] - Components::addComponent: reset orderLog, uid=22052!
  INFO root [2020-11-08 21:29:56,650] - Components::addComponent[logger], uid=22052, componentID=210002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2020-11-08 21:29:56,654] - ---- loginapp Version: 1.2.7. ScriptVersion: 0.1.0. Protocol: A87D1428E6DE7B1742DD55BD6E5BAE8C. Config: Debug X86. Built: 19:01:57 Aug  4 2020. AppID: 920002. UID: 22052. PID: 9880 ----
  INFO root [2020-11-08 21:29:56,655] - Server : Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz with 2097151kB RAM
  INFO root [2020-11-08 21:29:56,655] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2020-11-08 21:29:56,655] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [2020-11-08 21:29:56,791] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-11-08 21:29:56,791] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [2020-11-08 21:29:56,792] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [2020-11-08 21:29:56,796] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [2020-11-08 21:29:56,797] - Script::install_py_dlls(): PyInit_select
 DEBUG root [2020-11-08 21:29:56,797] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [2020-11-08 21:29:56,799] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [2020-11-08 21:29:56,840] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [2020-11-08 21:29:56,889] - Script::install(): is successfully, Python=(3.4.2 (default, Aug  4 2020, 19:01:54) [MSC v.1925 32 bit (Intel)])!
  INFO root [2020-11-08 21:29:56,889] - Script::install(): KBExtra is successfully!
S_INFO root [2020-11-08 21:29:56,892] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2020-11-08 21:29:56,892] - TelnetServer server is running on port 31000
  INFO root [2020-11-08 21:29:57,139] - ---- loginapp is running ----
 DEBUG root [2020-11-08 21:29:57,140] - Components::process(): Request for the process of identity...
  WARN root [2020-11-08 21:30:03,294] - Forwarding logs to logger[192.168.2.246:51760/0/0/0]...
 ERROR root [2020-11-08 21:46:23,621] - Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-11-08 21:46:23,623] - The following logs sent to logger failed:
  WARN root [2020-11-08 21:46:23,626] - ==> [2020-11-08 21:46:23 621] TCPPacketReceiver::processPendingEvents(192.168.2.246:51760): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-11-08 21:46:23,628] - ==> [2020-11-08 21:46:23 622] Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  WARN root [2020-11-08 21:46:23,683] - TCPPacketReceiver::processPendingEvents(192.168.2.246:51761): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2020-11-08 21:46:23,684] - Components::removeComponentByChannel: dbmgr : 42002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [2020-11-08 21:46:23,718] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2020-11-08 21:46:23,729] - onLoginAppShutDown()
  INFO root [2020-11-08 21:46:23,731] - Shutdowner::onShutdown: shutting down
  INFO root [2020-11-08 21:46:23,731] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [2020-11-08 21:46:23,735] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [2020-11-08 21:46:23,736] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [2020-11-08 21:46:23,737] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [2020-11-08 21:46:23,737] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [2020-11-08 21:46:23,737] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [2020-11-08 21:46:23,738] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [2020-11-08 21:46:23,738] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [2020-11-08 21:46:23,738] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [2020-11-08 21:46:23,769] - Script::uninstall(): is successfully!
 DEBUG root [2020-11-08 21:46:23,770] - ThreadPool::destroy(): starting size 2.
  WARN root [2020-11-08 21:46:24,097] - ThreadPool::destroy(): waiting for thread(2)[0x19dc050,0x19dc668,], try=1
